src/pages/workspaces/workspace/Data.js

```diff
@@ -129,216 +129,229 @@ const getReferenceData = _.flow(
+                    getOptionLabel: ({ value }) => _.startCase(value),
+                    value: editType,
+                    onChange: ({ value }) => setEditType(value),
+                    options: ['string', 'number', 'boolean', 'string list', 'number list', 'boolean list']
+                  }),
+                  h(Link, {
+                    tooltip: Utils.summarizeErrors(inputErrors) || 'Save changes',
+                    disabled: !!inputErrors.length,
+                    style: { marginLeft: '1rem' },
+                    onClick: () => saveAttribute(originalKey)
+                  }, [icon('success-standard', { size: 23 })]),
+                  h(Link, {
+                    tooltip: 'Cancel editing',
+                    style: { marginLeft: '1rem' },
+                    onClick: stopEditing
+                  }, [icon('times-circle', { size: 23 })])
+                ]) :
+                div({ className: 'hover-only' }, [
+                  h(Link, {
+                    disabled: !!Utils.editWorkspaceError(workspace),
+                    tooltip: Utils.editWorkspaceError(workspace) || 'Edit variable',
+                    style: { marginLeft: '1rem' },
+                    onClick: () => {
+                      setEditIndex(rowIndex)
+                      setEditValue(_.isObject(originalValue) ? originalValue.items.join(', ') : originalValue)
+                      setEditKey(originalKey)
+                      setEditType(_.isObject(originalValue) ? `${typeof originalValue.items[0]} list` : typeof originalValue)
+                    }
+                  }, [icon('edit', { size: 19 })]),
+                  h(Link, {
+                    'aria-label': 'Delete variable',
+                    disabled: !!Utils.editWorkspaceError(workspace),
+                    tooltip: Utils.editWorkspaceError(workspace) || 'Delete variable',
+                    style: { marginLeft: '1rem' },
+                    onClick: () => setDeleteIndex(rowIndex)
+                  }, [icon('trash', { size: 19 })])
+                ])
+            ])
+          }
+        }]
+      })])
+    ]),
+    !creatingNewVariable && editIndex === undefined && !Utils.editWorkspaceError(workspace) && h(FloatingActionButton, {
+      label: 'ADD VARIABLE',
+      iconShape: 'plus',
+      onClick: () => {
+        setEditIndex(filteredAttributes.length)
+        setEditValue('')
+        setEditKey('')
+        setEditType('string')
+      }
+    }),
+    deleteIndex !== undefined && h(Modal, {
+      onDismiss: () => setDeleteIndex(),
+      title: 'Are you sure you wish to delete this variable?',
+      okButton: h(ButtonPrimary, {
+        onClick: _.flow(
+          withErrorReporting('Error deleting workspace variable'),
+          Utils.withBusyState(setBusy)
+        )(async () => {
+          setDeleteIndex()
+          await Ajax().Workspaces.workspace(namespace, name).deleteAttributes([amendedAttributes[deleteIndex][0]])
+          loadAttributes()
```

Comment:
Since `loadAttributes` is writing to the same busy flag, this will cause the flag to get cleared early due to a race.
```suggestion
          await loadAttributes()
```